walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,87 @@
 
 package com.google.sps;
 
+import java.util.Arrays;
+import java.util.ArrayList;
+import java.util.Comparator;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
 
 public final class FindMeetingQuery {
+  /**
+   * @param events a collection of events going on during the day
+   * @param request user created request for an event that includes Time and attendees
+   * @return a list of available times to schedule a meeting
+   */
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    ArrayList<TimeRange> empty = new ArrayList<>();
+    ArrayList<String> attendees = new ArrayList<String>(request.getAttendees()); // list for mandatory attendees
+    ArrayList<String> allAttendees = new ArrayList<String>(attendees);
+    allAttendees.addAll(request.getOptionalAttendees());
+    // return whole day if there are no attendees
+    if (allAttendees.isEmpty()) {
+      return Arrays.asList(TimeRange.WHOLE_DAY);
+    } 
+    //return empty If there are no events during day(gaps), but there are optional attendees
+    if(events.isEmpty() && !request.getOptionalAttendees().isEmpty()){
```

Comment:
if events are empty, wouldn't the whole day be free then?

Reply:
according to the tutorial, no if there are optional attendees included to the meeting request

Reply:
add this edge case to javadoc